test/CKAN.jl

```diff
@@ -0,0 +1,11 @@
+using DataDepsGenerators
+using Base.Test
+
+using ReferenceTests
+
+@testset "CKAN test" begin
+    @test_reference "references/CKAN Gold Prices.txt" generate(CKAN(), "https://demo.ckan.org/api/3/action/package_show?id=gold-prices")
```

Comment:
Ideally,
I think the better interface would be to be calling:

```
@test_reference "references/CKAN Gold Prices.txt" generate(CKAN(), "https://demo.ckan.org/dataset/gold-prices")
````

and then transforming that URL into "https://demo.ckan.org/api/3/action/package_show?id=gold-prices" as part of the package.

I have a feeling the `find_metadata`  function needs reengineering.
Nowish is probably a good time to do that kind of refactor.